tests/scripts/run-throughput-perf.py

```diff
@@ -0,0 +1,361 @@
+
+    args = (arch, operating_system, build_type, run_type, clr_root, assembly_root, benchview_path)
+
+    # Log configuration
+    log('Configuration:')
+    log(' arch: %s' % arch)
+    log(' os: %s' % operating_system)
+    log(' build_type: %s' % build_type)
+    log(' run_type: %s' % run_type)
+    log(' clr_root: %s' % clr_root)
+    log(' assembly_root: %s' % assembly_root)
+    if not benchview_path is None:
+        log('benchview_path : %s' % benchview_path)
+
+    return args
+
+def nth_dirname(path, n):
+    """ Find the Nth parent directory of the given path
+    Args:
+        path (str): path name containing at least N components
+        n (int): num of basenames to remove
+    Returns:
+        outpath (str): path with the last n components removed
+    Notes:
+        If n is 0, path is returned unmodified
+    """
+
+    assert n >= 0
+
+    for i in range(0, n):
+        path = os.path.dirname(path)
+
+    return path
+
+def del_rw(action, name, exc):
+    os.chmod(name, stat.S_IWRITE)
+    os.remove(name)
+
+def log(message):
+    """ Print logging information
+    Args:
+        message (str): message to be printed
+    """
+
+    print('[%s]: %s' % (sys.argv[0], message))
+
+def generateCSV(dll_name, dll_runtimes):
+    """ Write throuput performance data to a csv file to be consumed by measurement.py
+    Args:
+        dll_name (str): the name of the dll
+        dll_runtimes (float[]): A list of runtimes for each iteration of the performance test
+    """
+
+    csv_file_name = "throughput_%s.csv" % (dll_name)
+    csv_file_path = os.path.join(os.getcwd(), csv_file_name)
+    output_file = open(csv_file_path, "w")
+
+    for iteration in dll_runtimes:
+        output_file.write("%s,%s,%s,%s\n" % ("default", "coreclr-crossgen-tp", dll_name, iteration))
+
+    output_file.close()
+    return csv_file_name
+
+def runIterations(dll_name, dll_path, iterations, crossgen_path, jit_path, assemblies_path):
+    """ Run throughput testing for a given dll
+    Args:
+        dll_name: the name of the dll
+        dll_path: the path to the dll
+        iterations: the number of times to run crossgen on the dll
+        crossgen_path: the path to crossgen
+        jit_path: the path to the jit
+        assemblies_path: the path to the assemblies that may be needed for the crossgen run
+    """
+
+    dll_runtimes = []
+
+    # Set up arguments for running crossgen
+    run_args = [crossgen_path,
+            '/JITPath',
+            jit_path,
+            '/Platform_Assemblies_Paths',
+            assemblies_path,
+            dll_path
+            ]
+
+    log(" ".join(run_args))
+
+    # Time.clock() returns seconds, with a resolution of 0.4 microseconds, so multiply by the multiplier to get milliseconds
+    multiplier = 1000
+
+    for iteration in range(0,iterations):
+        proc = subprocess.Popen(run_args, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)
```

Comment:
Reading through some of the Python docs on Popen they recommend not using shell=True, for security reasons.  Since we are not calling any shell specific things on Windows, and I don't believe that we need environment variable expansion we should be able to not do this.